backend/plan/views.py

```diff
@@ -1,19 +1,93 @@
+import os
+import pickle
+
 from django.core.exceptions import ObjectDoesNotExist
 from django.db import IntegrityError
 from django.db.models import Prefetch
 from django_auto_prefetching import AutoPrefetchViewSetMixin
 from rest_framework import status, viewsets
+from rest_framework.decorators import api_view, permission_classes, schema
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
 
 import plan.examples as examples
+from courses.management.commands.recommendcourses import (
+    clean_course_input,
+    recommend_courses,
+    vectorize_user,
+    vectorize_user_by_courses,
+)
 from courses.models import Section
 from courses.util import get_course_and_section, get_current_semester
 from PennCourses.docs_settings import PcxAutoSchema, reverse_func
+from PennCourses.settings.base import S3
 from plan.models import Schedule
 from plan.serializers import ScheduleSerializer
 
 
+def retrieve_course_clusters():
+    if not os.path.exists("./plan/temp_pickle/course-cluster-data.pkl"):
+        print("re-downloading...")
+
+        if "temp_pickle" not in os.listdir("./plan"):
+            os.mkdir("./plan/temp_pickle")
+
+        S3.download_file(
+            "penn.courses", "course-cluster-data.pkl", "./plan/temp_pickle/course-cluster-data.pkl"
+        )
+
+    return pickle.load(open("plan/temp_pickle/course-cluster-data.pkl", "rb"))
+
+
+@api_view(["POST"])
+@schema(PcxAutoSchema())
```

Comment:
Look at `PennCourses/docs_settings.py` for instructions on how to document this function-based view.